src/main/java/core/basesyntax/service/FruitShopServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.FruitShopDao;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.handler.OperationHandler;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class FruitShopServiceImpl implements FruitShopService {
+    private static final String DEFAULT_START = "fruit,quantity" + System.lineSeparator();
+    private static final String COMMA = ",";
+    private static final int INDEX_OF_OPERATION_TYPE = 0;
+    private static final int INDEX_OF_FRUIT = 1;
+    private static final int INDEX_OF_QUANTITY = 2;
+    private FruitShopDao fruitShopDao;
+    private OperationStrategy operationStrategy;
+
+    public FruitShopServiceImpl(FruitShopDao fruitShopDao, OperationStrategy operationStrategy) {
+        this.fruitShopDao = fruitShopDao;
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void addData(List<String> data) {
```

Comment:
This method is doing too much. It should not be used to parse string. Create Object with fields as in file type, name,amount. Create a separate parser service and parse string into that object. And here as parameter provide List of valid data

Reply:
I also do not fully understand this, is this neccessary?